test/linalg/dense.jl

```diff
@@ -185,354 +194,372 @@ for elty in (Int32, Int64, Float32, Float64, Complex64, Complex128)
+
+    # similar issue for Array{Real}
+    @test Real[1 2] * Real[1.5; 2.0] == Real[5.5]
+end
+
+@testset "Matrix exponential" begin
+    @testset "Tests for $elty" for elty in (Float32, Float64, Complex64, Complex128)
+        A1  = convert(Matrix{elty}, [4 2 0; 1 4 1; 1 1 4])
+        eA1 = convert(Matrix{elty}, [147.866622446369 127.781085523181  127.781085523182;
+                                     183.765138646367 183.765138646366  163.679601723179;
+                                     71.797032399996  91.8825693231832 111.968106246371]')
+        @test expm(A1) ≈ eA1
+
+        A2  = convert(Matrix{elty},
+                      [29.87942128909879    0.7815750847907159 -2.289519314033932;
+                       0.7815750847907159 25.72656945571064    8.680737820540137;
+                       -2.289519314033932   8.680737820540137  34.39400925519054])
+        eA2 = convert(Matrix{elty},
+                      [  5496313853692458.0 -18231880972009236.0 -30475770808580460.0;
+                       -18231880972009252.0  60605228702221920.0 101291842930249760.0;
+                       -30475770808580480.0 101291842930249728.0 169294411240851968.0])
+        @test expm(A2) ≈ eA2
+
+        A3  = convert(Matrix{elty}, [-131 19 18;-390 56 54;-387 57 52])
+        eA3 = convert(Matrix{elty}, [-1.50964415879218 -5.6325707998812  -4.934938326092;
+                                     0.367879439109187 1.47151775849686  1.10363831732856;
+                                     0.135335281175235 0.406005843524598 0.541341126763207]')
+        @test expm(A3) ≈ eA3
+
+        A4 = convert(Matrix{elty}, [0.25 0.25; 0 0])
+        eA4 = convert(Matrix{elty}, [1.2840254166877416 0.2840254166877415; 0 1])
+        @test expm(A4) ≈ eA4
+
+        A5 = convert(Matrix{elty}, [0 0.02; 0 0])
+        eA5 = convert(Matrix{elty}, [1 0.02; 0 1])
+        @test expm(A5) ≈ eA5
+
+        # Hessenberg
+        @test hessfact(A1)[:H] ≈ convert(Matrix{elty},
+                                                 [4.000000000000000  -1.414213562373094  -1.414213562373095
+                                                  -1.414213562373095   4.999999999999996  -0.000000000000000
+                                                  0  -0.000000000000002   3.000000000000000])
+    end
+
+    @testset "Additional tests for $elty" for elty in (Float64, Complex{Float64})
+
+        A4  = convert(Matrix{elty}, [1/2 1/3 1/4 1/5+eps();
+                                     1/3 1/4 1/5 1/6;
+                                     1/4 1/5 1/6 1/7;
+                                     1/5 1/6 1/7 1/8])
+        @test expm(logm(A4)) ≈ A4
+
+        A5  = convert(Matrix{elty}, [1 1 0 1; 0 1 1 0; 0 0 1 1; 1 0 0 1])
+        @test expm(logm(A5)) ≈ A5
+
+        A6  = convert(Matrix{elty}, [-5 2 0 0 ; 1/2 -7 3 0; 0 1/3 -9 4; 0 0 1/4 -11])
+        @test expm(logm(A6)) ≈ A6
+
+        A7  = convert(Matrix{elty}, [1 0 0 1e-8; 0 1 0 0; 0 0 1 0; 0 0 0 1])
+        @test expm(logm(A7)) ≈ A7
+
```

Comment:
Great, thanks! Fixed this and another code block where I had done the same thing.